### ISSUE TITLE
Using IsNull leads to comparison to null in generated SQL

### ISSUE THREAD

**Issue by bartlomiej-korpus** [typeorm/typeorm#4272]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

https://github.com/bartlomiej-korpus/typeorm-isnull-relations-bug

Hello!
So basically, when using .find with IsNull() on a field that also has the corresponding `fieldId` defined in the entity, the resulting SQL contains "= $1" where $1 is bound as null, instead of IS NULL.
That leads to quite surprising results.

Generated SQL is:
`SELECT "A"."id" AS "A_id", "A"."bId" AS "A_bId" FROM "a" "A" WHERE "A"."bId" = $1 -- PARAMETERS: [null]`

when it should be:

`query: SELECT "A"."id" AS "A_id", "A"."bId" AS "A_bId" FROM "a" "A" WHERE "A"."bId" IS NULL`

**Comment by vlapo** [comment]
Did not try it but we do not have any tests for relations and find operators. Is this planned in `next` @pleerock ?

**Comment by DTX-92** [comment]
Same issue.
Fixed by switching from a relation field to a table field:
`isNull(assignee) -> isNull(assigneeId)`

**Comment by 295** [comment]
@DTX-92 How did you solve it? I tried several ways and none of them worked.
```js
// 1
function getTreesViaParentId(parentId) {
    return getRepository(Tree).find({
        where: {
            parent: parentId === null ? IsNull() : { id: parentId }
        }
    })
}

// 2
function getTreesViaParentId(parentId) {
    return getRepository(Tree).find({
        where: {
            parent: { id: parentId === null ? IsNull() : parentId }
        }
    })
}
```
Generated SQL:
```sql
SELECT "Tree"."id" AS "Tree_id", "Tree"."parentId" AS "Tree_parentId" FROM "tree" "Tree" WHERE "Tree"."parentId" = $1 -- PARAMETERS: [null]
```
